### PR TITLE
External Input Devices Support

### PR DESCRIPTION
**Pull request by UelitonFreitas**
Add external input devices support  such as TV's remote controls, Video Game Controls and others.
Initialy support `PLAY`, `PAUSE`, `STOP` and `PLAY_PAUSE` buttons.

### CODE REVIEW

File: clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt [hunk 1]
```diff
@@ -321,110 +324,127 @@ open class PlayerTest {
...
 
     @Test
-    fun shouldListenExitFullscreenEventOutOfPlayerTriggeredByCore() {
+    fun `should listen exit fullscreen event out of player triggered by core`() {
         assertCoreEventWasTriggered(Event.EXIT_FULLSCREEN.value)
     }
 
+    @Test
+    fun `should pass key event to external input class`(){
+
```
Comment (arthursz) [inline clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt]: Can we remove this empty line, please?

File: clappr/src/main/kotlin/io/clappr/player/base/keys/Key.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.clappr.player.base.keys
+
+enum class Key(val value: String) {
+    UNDEFINED("undefined"),
+    PLAY("play"),
+    PAUSE("pause"),
+    STOP("strop"),
```
[the remark this answers is not in the document]
Comment (UelitonFreitas) [reply clappr/src/main/kotlin/io/clappr/player/base/keys/Key.kt]: Sure. Thank you. Done 🤘

File: clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt (hunk 1 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt]: Nice. Thank you and done!

File: clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt [hunk 3]
```diff
@@ -567,4 +586,19 @@ open class PlayerTest {
...
+            override val name = ExternalInputPlugin.name
+
+            val entry = PluginEntry.Core(name = name, factory = { core -> ExternalInputPluginTest(core) })
+
+            var keyEvent: KeyEvent? = null
+        }
+
+        override fun holdKeyEvent(event: KeyEvent) {
+            keyEvent = event
+
```
Comment (yaliscouto) [inline clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt]: unnecessary line

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    @Before
+    fun setUp() {
+        BaseObject.applicationContext = ApplicationProvider.getApplicationContext()
+
+        core = Core(Options())
+
+        externalInputPlugin = ExternalInputPlugin(core)
+    }
+
+
```
Comment (yaliscouto) [inline clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: unnecessary line

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    fun `should trigger KEY_PRESSED event for PLAY button`() {
+        assertPressedKeyCodeEvent(Key.PLAY, KeyEvent.KEYCODE_MEDIA_PLAY)
+    }
+
```
Comment (yaliscouto) [inline clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: unnecessary line

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    private fun assertPressedKeyCodeEvent(expectedKeyCode: Key, keyToHold: Int){
+        var keyCode: String? = null
+
+        core.on(Event.DID_RECEIVE_INPUT_KEY.value) { bundle ->
+            bundle?.let { keyCode = it.getString(EventData.INPUT_KEY_CODE.value) }
+        }
+
+        externalInputPlugin.holdKeyEvent(KeyEvent(-1, keyToHold))
```
Comment (yaliscouto) [inline clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: what is this value -1? Can we put it into a constant?

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    private fun assertPressedAction(expectedActionCode: Action, actionKey: Int){
+        var actionCode: String? = null
+
+        core.on(Event.DID_RECEIVE_INPUT_KEY.value) { bundle ->
+            bundle?.let { actionCode = it.getString(EventData.INPUT_KEY_ACTION.value) }
+        }
+
+        externalInputPlugin.holdKeyEvent(KeyEvent(actionKey, -1))
```
Comment (yaliscouto) [inline clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: same here

File: clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt (hunk 3 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/PlayerTest.kt]: Nice. Done!

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt (hunk 4 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: Done!

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt (hunk 5 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: Done!

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt (hunk 6 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: Sure. Done!

File: clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt (hunk 7 above)
Comment (UelitonFreitas) [reply clappr/src/test/kotlin/io/clappr/player/plugin/ExternalInputPluginTest.kt]: Done. 🤘

**Pull request merged**